spec/features/create_short_url_spec.rb

```diff
@@ -2,13 +2,60 @@
 
 describe "creating a short url" do
   context "when not signed in" do
-  end
+    let(:url) { "http://www.microsoft.com" }
+
+    before do
+      visit "/home"
+      fill_in "link[long_url]", with: url
+      click_button "Shorten It!"
+    end 
+
+    it "should have a short link" do
+      expect(page).to have_content /shrt.nr\/[a-z0-9]{6}/
+    end 
+
+    it "should tell where the short link points" do
+       expect(page).to have_content url
+    end 
+
+  end 
 
   context "when signed in" do
     let(:user) { create(:user) }
-
+    let(:url) { "http://www.microsoft.com" }
+    
     before do
       login_as user
+      fill_in "link[long_url]", with: url
+      click_button "Shorten It!"
+    end 
+
+    it "should have a short link" do
+      expect(page).to have_content /shrt.nr\/[a-z0-9]{6}/
+    end 
+
+    it "should tell where the short link points" do
+       expect(page).to have_content url
+    end 
+
+    it "should have the link in the database" do
```

Comment:
Tests that need to query the database are best found in functional tests like spec/controllers/links_controller_spec.rb

Feature tests should only test what the User can interact with